### PR TITLE
[Fix #5800] Replace backtrace with --help suggestion for invalid options

### LINKED ISSUE #5800: Don't show a stracktrace for invalid command-line params
## Expected behavior

It'd be nice to get some helpful message when you mistype some CLI param.

## Actual behavior

You get an ugly scary stacktrace.

```
rubocop --show-copss
invalid option: --show-copss
/Users/bozhidar/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rubocop-0.55.0/lib/rubocop/options.rb:21:in `parse'
/Users/bozhidar/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rubocop-0.55.0/lib/rubocop/cli.rb:37:in `run'
/Users/bozhidar/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rubocop-0.55.0/bin/rubocop:13:in `block in <top (required)>'
/Users/bozhidar/.rbenv/versions/2.4.1/lib/ruby/2.4.0/benchmark.rb:308:in `realtime'
/Users/bozhidar/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rubocop-0.55.0/bin/rubocop:12:in `<top (required)>'
/Users/bozhidar/.rbenv/versions/2.4.1/bin/rubocop:22:in `load'
/Users/bozhidar/.rbenv/versions/2.4.1/bin/rubocop:22:in `<main>'
```

## Steps to reproduce the problem

Just type something like `rubocop --foo`.  

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.55.0 (using Parser 2.5.0.0, running on ruby 2.4.2 x86_64-linux)
```

### PR DESCRIPTION
**Pull request by shanecav84**
Removes the backtrace printed when invalid options are passed to the CLI, and shows a suggestion to use the `--help` flag instead.

Closes #5800.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `rake default` or `rake parallel`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by bbatsov** [comment]
Can you please add one unit test about the change?

File: lib/rubocop/cli.rb [hunk 1]
```diff
@@ -52,7 +52,7 @@ def run(args = ARGV)
       STATUS_ERROR
     rescue StandardError, SyntaxError, LoadError => e
       warn e.message
-      warn e.backtrace
+      warn 'For usage information, use --help'
```
Comment (Drenmi) [inline lib/rubocop/cli.rb]: This would remove the stack trace for a host of other issues though, in addition to invalid command line parameters?

Comment (shanecav84) [reply lib/rubocop/cli.rb]: Good catch. Will update.

**Comment by shanecav84** [comment]
Spec is up.

**Comment by bbatsov** [comment]
Seems the CI's failing.

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**